File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/jms/docs/configuration/ActiveMQXAConnectionBeanFactory.kt
package io.micronaut.jms.docs.configuration

import io.micronaut.context.annotation.Requires
// tag::imports[]
import io.micronaut.context.annotation.Factory
import io.micronaut.jms.activemq.classic.configuration.properties.ActiveMqClassicConfigurationProperties
import io.micronaut.jms.annotations.JMSConnectionFactory
import jakarta.inject.Singleton
import org.apache.activemq.ActiveMQXAConnectionFactory

// end::imports[]
@Requires(property = "spec.name", value = "CustomBrokerSpec") // tag::clazz[]
// end::clazz[]
@Factory
internal class ActiveMQXAConnectionBeanFactory {
    @JMSConnectionFactory("activeMQXAConnectionFactory")
    @Singleton
    fun createActiveMQXAConnectionFactory(amqConfig: ActiveMqClassicConfigurationProperties): ActiveMQXAConnectionFactory {
        return ActiveMQXAConnectionFactory(amqConfig.getConnectionString())
    }
} // end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/jms/docs/configuration/ActiveMQXAConnectionBeanFactory.py
# tag::imports[]
from jakarta.inject import Singleton
from micronaut.context.annotation import Factory, Requires
from micronaut.jms.activemq.classic.configuration.properties import ActiveMqClassicConfigurationProperties
from micronaut.jms.annotations import JMSConnectionFactory
from org.apache.activemq import ActiveMQXAConnectionFactory

# end::imports[]


@Requires(property="spec.name", value="CustomBrokerSpec")
# tag::clazz[]
@Factory
class ActiveMQXAConnectionBeanFactory:

    @JMSConnectionFactory("activeMQXAConnectionFactory")
    @Singleton
    def create_active_mq_xa_connection_factory(self, amq_config: ActiveMqClassicConfigurationProperties) -> ActiveMQXAConnectionFactory:
        return ActiveMQXAConnectionFactory(amq_config.getConnectionString())
# end::clazz[]
